modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Filters.tsx

```diff
@@ -39,6 +42,87 @@ interface Filter {
 	type: string;
 }
 
+interface DateFilter extends Filter {
+	from: string;
+	to: string;
+}
+
+function DateRange({
+	from,
+	namespace,
+	onFromChange,
+	onToChange,
+	to,
+}: {
+	from: string;
+	namespace: string;
+	onFromChange: (from: string) => void;
+	onToChange: (to: string) => void;
+	to: string;
+}) {
+	const [isValid, setIsValid] = React.useState(true);
+
+	React.useEffect(() => {
+		if (from && to) {
+			setIsValid(isBefore(new Date(from), new Date(to)));
+		}
+	}, [from, to]);
```

Comment:
Will `to` and `from` ever change? If not, and the state will never change internally, I believe the best practice here is to use `useRef` to store `isValid`.

Reply:
Yeah from/to will change when someone selects a new value in the date picker, which is why we need to re-validate it when they change.

Hmm, not sure about a "best practice" to use ref's like that, or at least not that I am aware of. Here is some info from the docs.
<img width="791" alt="Screen Shot 2023-05-05 at 4 02 14 PM" src="https://user-images.githubusercontent.com/6843530/236452566-ee8a228a-9883-41bd-a137-dfc4f4827719.png">

Typically I usually see refs for element  references. If it was just to hold a static value, we can declare them outside the component as a variable.

Reply:
Greg drilled this as best practice to me, in several reviews like https://github.com/wincent/liferay-portal/pull/88#discussion_r340560057. Basically, `ref` results in better performance, if we know value will be stable after initial render.

Reply:
hmm. That seems pretty odd to me, I don't think I would agree with that. I haven't really seen that pattern used in React(using ref for form/input values), I typically see input values read from either a controlled input with state or rely on form submission and let the html handle the value. The performance improvement seems pretty negligible, otherwise I think the react docs would be promoting this more under forms.

From my understanding, and from what I have seen and read in the docs, ref is typically used to store a value that is _unrelated_ to the render lifecycle. 

-----

And actually, after seeing the comment Greg made, I think he was actually talking about something slightly different. He was recommending to use ref because `setEditorTextInput` is only used once, and it is when `A.AceEditor` fires the "render" event, which only happens once and isn't tied into the render lifecycle. If `setEditorTextInput` was used in the React component's lifecycle, then it would make sense to keep using state instead of ref.

Reply:
> From my understanding, and from what I have seen and read in the docs, ref is typically used to store a value that is unrelated to the render lifecycle.

Well, yes, that's why I was asking if the `to` and `from` props will ever change. As they will, we need to use `useEffect`, we cannot use `ref`. All good. If they never changed, and state never changed internally, then they would be unrelated to the render lifecycle.